AIO/src/org/aio/activities/quests/RomeoAndJuliet.java

```diff
@@ -150,16 +147,27 @@ private void talkToJuliet() throws InterruptedException {
         }
     }
 
-    private void getItemFromObject(Area place, String itemName, String objectName, String interaction) throws InterruptedException {
-        if (place.contains(myPlayer())) {
-            RS2Object object = getObjects().closest(o -> o.getName().equals(objectName) && o.hasAction(interaction));
+    private void getBerry(){
+        String interaction = "Pick-from";
+        if(BERRIES.contains(myPlayer())){
+            List<RS2Object> objects = getObjects().filter(o -> o.getName().equals("Cadava bush") && o.hasAction(interaction));
+            //Antipattern pick a random Bush
+            RS2Object object = objects.get(random(0,objects.size()-1));
             if (object != null && object.interact(interaction)) {
-                Sleep.sleepUntil(() -> getInventory().contains(itemName), 15000);
-            }
-        } else {
-            getWalking().webWalk(place);
-        }
+                Sleep.sleepUntil(() -> {
+                        if(getTabs().isOpen(Tab.INVENTORY)){
```

Comment:
We don't need to open the inventory tab to check if we have an item.

Reply:
In my testing i needed to, otherwise he would pick infinite 🤷‍♂️

Reply:
The inventory should open, if it's not open. As seen on line 56.